app/jobs/export_tags_job.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+##
+# A job that exports tags to CSV for one or more objects
+# @param [Integer] bulk_action_id GlobalID for a BulkAction object
+# @param [Hash] _params additional parameters that an Argo job may need
+class ExportTagsJob < GenericJob
+  def perform(bulk_action_id, _params)
+    super
+
+    with_bulk_action_log do |log_buffer|
+      log_buffer.puts("#{Time.current} Starting #{self.class} for BulkAction #{bulk_action_id}")
+      update_druid_count
+
+      csv_string = CSV.generate do |csv|
+        pids.each do |druid|
+          log_buffer.puts("#{Time.current} #{self.class}: Exporting tags for #{druid} (bulk_action.id=#{bulk_action_id})")
+          csv << [druid, *export_tags(druid)]
+          bulk_action.increment(:druid_count_success).save
+        rescue StandardError => e
+          log_buffer.puts("#{Time.current} #{self.class}: Unexpected error exporting tags for #{druid} (bulk_action.id=#{bulk_action.id}): #{e}")
+          bulk_action.increment(:druid_count_fail).save
+        end
+      end
+
+      File.open(csv_download_path, 'w') { |file| file.write(csv_string) } if csv_string.present?
```

Comment:
Why are we writing the CSV to memory instead of straight to file?

Reply:
Damn good question. I'll make that change.

Reply:
@justinlittman take another look? thanks for the suggestion!